packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/common/Helpers.cs

```diff
@@ -15,56 +13,36 @@ public static class Helpers
     {
         private static readonly string _assemblyLocation = Path.GetDirectoryName(typeof(Helpers).Assembly.Location)!;
 
-        public static string GetExpectedFromFile(string? parameters = null)
+        public static string GetExpectedFromFile(
+            string? parameters = null,
+            [CallerMemberName] string method = "",
+            [CallerFilePath] string filePath = "")
         {
-            return File.ReadAllText(GetAssetFileOrDirectoryPath(true, parameters));
+            return File.ReadAllText(GetAssetFileOrDirectoryPath(true, parameters, method, filePath));
         }
 
-        private static string GetAssetFileOrDirectoryPath(bool isFile, string? parameters = null)
+        public static string GetAssetFileOrDirectoryPath(
+            bool isFile,
+            string? parameters = null,
+            [CallerMemberName] string method = "",
+            [CallerFilePath] string filePath = "")
         {
-            var stackTrace = new StackTrace();
-            var stackFrame = GetRealMethodInvocation(stackTrace);
-            var method = stackFrame.GetMethod();
-            var callingClass = method!.DeclaringType;
-            var nsSplit = callingClass!.Namespace!.Split('.');
+            var pieces = filePath.Split('\\');
+            var callingClass = pieces.Last().Split('.').First();
             var paramString = parameters is null ? string.Empty : $"({parameters})";
             var extName = isFile ? ".cs" : string.Empty;
-            var path = _assemblyLocation;
-            var nsSkip = nsSplit.Contains("ClientModel") ? 5 : 4;
-            for (int i = nsSkip; i < nsSplit.Length; i++)
-            {
-                path = Path.Combine(path, nsSplit[i]);
-            }
-            return Path.Combine(path, "TestData", callingClass.Name, $"{method.Name}{paramString}{extName}");
-        }
-
-        private static StackFrame GetRealMethodInvocation(StackTrace stackTrace)
```

Comment:
This was not always accurate under certain circumstances. I've switched to using the caller attributes.